### PR TITLE
HPCC-29674 LogAccess support for pod column type

### PR DESCRIPTION
**Pull request by rpastrana**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
image

**Review by jakesmith — COMMENTED** [review]
@rpastrana - All looks reasonable, 1 minor comment re. log level.

Will revisit to approve when base PR  is merged.
Please tag me here when that is merged and this PR is rebased.

**Comment by rpastrana** [comment]
@jakesmith rebased.

**Comment by rpastrana** [comment]
@jeclrsg does this provide everything we need for https://track.hpccsystems.com/browse/HPCC-30568?

**Review by jeclrsg — APPROVED** [review]
@rpastrana I think so. Looks good to me.

File: system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp [hunk 1]
```diff
@@ -725,6 +736,19 @@ void AzureLogAnalyticsCurlClient::populateKQLQueryString(StringBuffer & queryStr
...
+    {
+        queryField = m_podSearchColName.str();
+
+        if (!m_podIndexSearchPattern.isEmpty())
+        {
+            throwIfMultiIndexDetected(queryIndex.str(),  m_podIndexSearchPattern.str());
+            queryIndex = m_podIndexSearchPattern.str();
+        }
+
+        DBGLOG("%s: Searching log entries by Pod: '%s'", COMPONENT_NAME, queryValue.str() );
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp:749]: I do agree - but would require a more comprehensive review of all DBGLOG's I suspect before we commit to turning them off by default.

Is there a JIRA this topic?

**Review by jakesmith — APPROVED** [review]
@rpastrana - looks good.

File: system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (rpastrana) [reply system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp:749]: Yep, in fact we have too many jiras related to this topic:

HPCC-22646  Revisit logging levels
HPCC-18399 	Use jlog logging more consistently 
HPCC-24433 Log levels above max threshold should be reevaluated 
HPCC-30629 	Clean up roxie logs 
HPCC-26886 	Consider adding detail level to the log output 
HPCC-22650  Switch to using the tracing levels in logging

**Pull request merged**